gulpfile.js

```diff
@@ -1,43 +1,45 @@
 var gulp = require("gulp");
 var uglify = require("gulp-uglify");
-var rename = require("gulp-rename");
-var cleanCSS = require("gulp-clean-css");
+var concat = require("gulp-concat");
+var sass = require("gulp-sass");
 
 var config = {};
 config.srcPath = "./src/";
 config.distPath = "./dist/";
+config.vendorPath = "./vendor/";
 
-config.cssSrcPath = config.srcPath + "css/*.css";
+config.sassSrcPath = config.srcPath + "sass/*.scss";
 config.jsSrcPath = config.srcPath + "js/*.js";
 
-config.cssDistPath = config.distPath + "css/";
-config.jsDistPath = config.distPath + "js/";
+// JS files to concatenate
+config.jsFilesToConcat = [
+    // The order of files here *matters*
+    config.vendorPath + "form.js",

```

Comment:
Instead of explicitly defining filenames, define paths so that new files can be processed without making changes in gulpfile.

Reply:
You can't because **The order of files here matters** when they are being concatenated.

Globing the directory and concatenating can lead to unexpected behaviour.

For instance you don't want to include `form.js` after it is being called in `main.js` as a plugin.

I think it's better to define a file to be included instead of dropping it into a folder and *hoping* the task handles it.

If you are sure you want me to re-write this, let me know.

Reply:
`gulp.src` emits files in order they were added. Here is the [source](http://www.competa.com/blog/how-to-add-multiple-directories-in-one-gulp-task/). Could you please test it?

Reply:
Think I misunderstood, is this now what you were after?
Pleas let me know.